Oddly this failed on CI though:

```
11:35:34 remote: ---> Starting build... but first, coffee!        
11:35:34 remote: Step 1 : FROM alpine:3.3        
11:35:34 remote: ---> 6c2aa2137d97        
11:35:34 remote: Step 2 : RUN apk add -U 	bash 	nginx 	&& rm -rf /var/cache/apk*        
11:35:34 remote: ---> Using cache        
11:35:34 remote: ---> daf48dd65de3        
11:35:34 remote: Step 3 : RUN ln -sf /dev/stdout /var/log/nginx/access.log        
11:35:34 remote: ---> Using cache        
11:35:34 remote: ---> 056a79dfd1ff        
11:35:34 remote: Step 4 : RUN ln -sf /dev/stderr /var/log/nginx/error.log        
11:35:34 remote: ---> Using cache        
11:35:34 remote: ---> 1b9e99e78554        
11:35:34 remote: Traceback (most recent call last):        
11:35:34 remote:   File "/deploy.py", line 103, in <module>        
11:35:34 remote:     log_output(stream, True)        
11:35:34 remote:   File "/deploy.py", line 25, in log_output        
11:35:34 remote:     print(stream_chunk.replace(b'\n', b'').decode('utf-8'))        
11:35:34 remote: UnicodeEncodeError: 'ascii' codec can't encode characters in position 18-27: ordinal not in range(128) 
```

--- a/rootfs/Dockerfile
+++ b/rootfs/Dockerfile
@@ -1,13 +1,13 @@
 FROM quay.io/deis/base:v0.3.5
 
-RUN buildDeps='gcc git libffi-dev libssl-dev python-dev python-pip python-wheel python-setuptools'; \
+RUN buildDeps='gcc git libffi-dev libssl-dev python3-dev python3-pip python3-wheel python3-setuptools'; \
     apt-get update && \
     apt-get install -y --no-install-recommends \
         $buildDeps \
         libffi6 \
         libssl1.0.0 \
-        python && \
-	pip install --disable-pip-version-check --no-cache-dir docker-py==1.10.6 && \
+        python3-minimal && \
+	pip3 install --disable-pip-version-check --no-cache-dir docker-py==1.10.6 && \
     # cleanup
     apt-get purge -y --auto-remove $buildDeps && \
     apt-get autoremove -y && \
@@ -35,4 +35,4 @@ RUN chmod +x /bin/objstorage
 
 COPY . /
 
-CMD ["python", "/deploy.py"]
+CMD ["python3", "/deploy.py"]

--- a/rootfs/deploy.py
+++ b/rootfs/deploy.py
@@ -1,5 +1,6 @@
 import docker
 import os
+import sys
 import tarfile
 import time
 import requests
@@ -12,19 +13,21 @@ registryLocation = os.getenv('DEIS_REGISTRY_LOCATION', 'on-cluster')
 def log_output(stream, decode):
     error = False
     for chunk in stream:
+        if isinstance(chunk, bytes):
+            # Convert to dict, since docker-py returns some errors as raw bytes.
+            chunk = eval(chunk)
         if 'error' in chunk:
             error = True
-            if isinstance(chunk, basestring):  # Change "basestring" to "str" for Python3
-                print(chunk.decode('utf-8'))
-            else:
-                print(chunk['error'].decode('utf-8'))
+            print(chunk['error'])
         elif decode:
             stream_chunk = chunk.get('stream')
             if stream_chunk:
-                stream_chunk = stream_chunk.encode('utf-8').strip()
-                print(stream_chunk.replace('\n', ''))
+                # Must handle chunks as bytes to avoid UnicodeEncodeError.
+                encoded_chunk = stream_chunk.encode('utf-8')
+                sys.stdout.buffer.write(encoded_chunk)
         elif DEBUG:
-            print(chunk.decode('utf-8'))
+            print(chunk)
+        sys.stdout.flush()
     if error:
         # HACK: delay so stderr is logged before this dockerbuilder pod exits.
         time.sleep(3)

